app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts

```diff
@@ -16,42 +28,59 @@ import { CaveatTypes } from '../../../../../shared/constants/permissions';
 import { UNKNOWN_TICKER_SYMBOL } from '../../../../../shared/constants/app';
 import { PermissionNames } from '../../../controllers/permissions';
 import { getValidUrl } from '../../util';
+import {
+  EndApprovalFlow,
+  GetCaveat,
+  GetChainPermissionsFeatureFlag,
+  RequestPermittedChainsPermission,
+  RequestUserApproval,
+  SetActiveNetwork,
+} from './types';
 
-export function findExistingNetwork(chainId, findNetworkConfigurationBy) {
+export function findExistingNetwork(
+  chainId: Hex,
+  findNetworkConfigurationBy: (options: { chainId: Hex }) => void,
+) {
   if (
     Object.values(BUILT_IN_INFURA_NETWORKS)
-      .map(({ chainId: id }) => id)
+      .map(({ chainId: id }) => id as Hex)
       .includes(chainId)
   ) {
     return {
       chainId,
       ticker: CURRENCY_SYMBOLS.ETH,
-      nickname: NETWORK_TO_NAME_MAP[chainId],
-      rpcUrl: CHAIN_ID_TO_RPC_URL_MAP[chainId],
-      type: CHAIN_ID_TO_TYPE_MAP[chainId],
+      nickname:
+        NETWORK_TO_NAME_MAP[chainId as keyof typeof NETWORK_TO_NAME_MAP],
+      rpcUrl:
+        CHAIN_ID_TO_RPC_URL_MAP[
+          chainId as keyof typeof CHAIN_ID_TO_RPC_URL_MAP
+        ],
+      type: CHAIN_ID_TO_TYPE_MAP[chainId as keyof typeof CHAIN_ID_TO_TYPE_MAP],
     };
   }
   return findNetworkConfigurationBy({ chainId });
 }
 
-export function validateChainId(chainId) {
+export function validateChainId(chainId: Hex) {
   const _chainId = typeof chainId === 'string' && chainId.toLowerCase();
   if (!isPrefixedFormattedHexString(_chainId)) {
     throw ethErrors.rpc.invalidParams({
       message: `Expected 0x-prefixed, unpadded, non-zero hexadecimal string 'chainId'. Received:\n${chainId}`,
     });
   }
 
-  if (!isSafeChainId(parseInt(_chainId, 16))) {
+  if (!isSafeChainId(parseInt(_chainId as Hex, 16))) {
     throw ethErrors.rpc.invalidParams({
       message: `Invalid chain ID "${_chainId}": numerical value greater than max safe value. Received:\n${chainId}`,
     });
   }
 
-  return _chainId;
+  return _chainId as Hex | false;
 }
 
-export function validateSwitchEthereumChainParams(req, end) {
+export function validateSwitchEthereumChainParams<
+  Params extends JsonRpcParams = JsonRpcParams,
+>(req: JsonRpcRequest<Params>) {
```

Comment:
Since we're assuming that params will contain `chainId` with the destructuring statement, this should be a safe assumption. 

It also removes the need for the as assertion in the return statement.

```suggestion
export function validateSwitchEthereumChainParams<
  Params extends JsonRpcParams = { chainId: Hex } & JsonRpcParams,
>(req: JsonRpcRequest<Params>) {
```